### PR TITLE
openlens: init at 6.4.5

### PR DESCRIPTION
**Pull request by benwbooth**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/openlens/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Icon=open-lens' 'Icon=${pname}' \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+    '';
+
+  meta = with lib; {
+    description = "The Kubernetes IDE";
+    homepage = "https://github.com/MuhammedKalkan/OpenLens";
+    license = licenses.mit;
+    maintainers = with maintainers; [ benwbooth ];
```
Comment (wegank) [inline pkgs/applications/networking/cluster/openlens/default.nix:39]: Please add yourself to the maintainer list in a separate commit: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

Comment (benwbooth) [reply pkgs/applications/networking/cluster/openlens/default.nix:39]: Done. Thanks!

**Comment by wegank** [comment]
I just saw the last commit, and I wonder what is the point of going to an alpha version, when 6.3.0 was released three weeks ago...

**Comment by benwbooth** [comment]
6.3.0 is a controversial release that removes a lot of features. The 6.4.0-alpha.2 brings back some of the removed features. See https://github.com/lensapp/lens/issues/6823

**Comment by wegank** [comment]
Ok, thanks for the information!

**Comment by benwbooth** [comment]
I had to revert to version 6.2.5 since that is the most recent non-broken version before 6.3.0.

**Comment by SebTM** [comment]
Thanks for the PR, I tried it and had some issues (I suspect Wayland?) with the following error-message:

```
(open-lens:2): Gdk-ERROR **: 19:43:24.545: The program 'open-lens' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadAccess (attempt to access private resource denied)'.
  (Details: serial 251 error_code 10 request_code 130 (MIT-SHM) minor_code 1)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```

With some help and context from #104412, #109466 I found a solution: `unshareIpc = false` inside `appimageTools.wrapType2`-block. Exporting `QT_X11_NO_MITSHM=1` before launching openlens from console didn't helped, unsharePid seems to be not necessary for me.

**Comment by benwbooth** [comment]
@SebTM are you using X11 or wayland? I'm using wayland and haven't encountered this error.

**Comment by SebTM** [comment]
I'm using Wayland, with sway - I'm on unstable `e1e1b192c1a5aab2960bf0a0bd53a2e8124fa18e`

**Comment by SebTM** [comment]
Hey, thanks for following up - shortly after there was also v6.4.3 released. If you like to, you can add me to maintainer as well. Please update the title and check-boxes when done accordingly and I would think we are good-2-go ✌🏻

**Review by SebTM — APPROVED** [review]

**Comment by benwbooth** [comment]
@SebTM I updated to 6.4.3 and added you as a maintainer. Thanks!

**Review by SebTM — APPROVED** [review]

**Comment by SebTM** [comment]
@benwbooth Can't merge myself, so mentioned it as reviewed in a discourse thread - so nothing wrong just waiting for some attention ✌🏻

File: pkgs/applications/networking/cluster/openlens/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib, fetchurl, appimageTools, wrapGAppsHook }:
+
+let
+  pname = "openlens";
+  version = "6.4.5";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/openlens/default.nix:6]:
```suggestion
```

File: pkgs/applications/networking/cluster/openlens/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    name = "${pname}.AppImage";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+
+in
+appimageTools.wrapType2 {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/openlens/default.nix:20]:
```suggestion
  inherit pname version src;
```

File: pkgs/applications/networking/cluster/openlens/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      mv $out/bin/${name} $out/bin/${pname}
+
+      install -m 444 -D ${appimageContents}/open-lens.desktop $out/share/applications/${pname}.desktop
+      install -m 444 -D ${appimageContents}/usr/share/icons/hicolor/512x512/apps/open-lens.png \
+         $out/share/icons/hicolor/512x512/apps/${pname}.png
+
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Icon=open-lens' 'Icon=${pname}' \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+    '';
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/openlens/default.nix:34]:
```suggestion
  extraInstallCommands = ''
    mv $out/bin/${pname}-${version} $out/bin/${pname}

    install -m 444 -D ${appimageContents}/open-lens.desktop $out/share/applications/${pname}.desktop
    install -m 444 -D ${appimageContents}/usr/share/icons/hicolor/512x512/apps/open-lens.png \
       $out/share/icons/hicolor/512x512/apps/${pname}.png

    substituteInPlace $out/share/applications/${pname}.desktop \
      --replace 'Icon=open-lens' 'Icon=${pname}' \
      --replace 'Exec=AppRun' 'Exec=${pname}'
  '';
```

**Review by SebTM — APPROVED** [review]

**Comment by SebTM** [comment]
I guess you should squash the commits together that it's only one or if possible check that it's enabled on pr-side to do so on merge :v:

**Comment by benwbooth** [comment]
@SebTM I squashed the commits. Thanks!

**Review by SebTM — APPROVED** [review]

**Pull request merged**